### PR TITLE
Editor Store: Fix refreshPost() effect

### PR DESCRIPTION
**Pull request by ockham**
## Description
Found while working on https://github.com/Automattic/wp-calypso/pull/28549. `refreshPost()` uses `data` for `context: 'edit'`, even though it is a `GET` request (`method` isn't specified, thus falling back to `GET`). This causes `window.fetch()` to fail.

The fix is to use a query arg instead.

### To reproduce the bug on `master`:

- Edit a post
- In the browser console, enter `wp.data.dispatch( 'core/editor' ).refreshPost()`
- This triggers the following error:

```
Uncaught (in promise)
{code: "invalid_json", message: "The response is not a valid JSON response."}
```

### To reproduce the fix

Repeat the same steps. Verify that no error is triggered.

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation.

### CODE REVIEW

**Review by tyxla — APPROVED** [review]
This works well, LGTM 👍

File: packages/editor/src/store/effects/posts.js [hunk 1]
```diff
@@ -313,8 +314,10 @@ export const refreshPost = async ( action, store ) => {
 	const postTypeSlug = getCurrentPostType( getState() );
 	const postType = await resolveSelector( 'core', 'getPostType', postTypeSlug );
 	const newPost = await apiFetch( {
-		path: `/wp/v2/${ postType.rest_base }/${ post.id }`,
-		data: { context: 'edit' },
+		path: addQueryArgs(
+			`/wp/v2/${ postType.rest_base }/${ post.id }`,
```
Comment (youknowriad) [inline packages/editor/src/store/effects/posts.js]: I don't mind just concatenating the query string :)

**Review by youknowriad — APPROVED** [review]
LGTM 👍

File: packages/editor/src/store/effects/posts.js (hunk 1 above)
Comment (ockham) [reply packages/editor/src/store/effects/posts.js]: Yeah, `addQueryArgs` might be overblown for something entirely static. I'll drop it.

**Comment by ockham** [comment]
Thanks a lot @youknowriad!

**Pull request merged**